py4DSTEM/process/preprocess/preprocess.py

```diff
@@ -149,5 +150,117 @@ def bin_data_real(datacube, bin_factor):
+    return_mask=False
+    ):
+    """
+    This function performs pixel filtering to remove hot / bright pixels. We first compute a moving local ordering filter,
+    applied to the mean diffraction image. This ordering filter will return a single value from the local sorted intensity 
+    values, given by ind_compare. ind_compare=0 would be the highest intensity, =1 would be the second hightest, etc.
+    Next, a mask is generated for all pixels which are least a value thresh higher than the local ordering filter output.
+    Finally, we loop through all diffraction images, and any pixels defined by mask are replaced by their 3x3 local median.
+
+    Args:
+            datacube (DataCube):      py4DSTEM Datacube
+            thresh (float):           threshold for replacing hot pixels, if pixel value minus local ordering filter exceeds it.
+            ind_compare (int):        which median filter value to compare against. 0 = brightest pixel, 1 = next brightest, etc.
+
+        Returns:
+            datacube                     datacube              
+            mask (bool):                 (optional) the bad pixel mask
+    """
+
+    # Mean image over all probe positions
+    diff_mean = np.mean(datacube.data,axis=(0,1))
+
+    # Local median
+    diff_local_med = np.sort(np.vstack([
+        np.roll(diff_mean,(-1,-1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 0,-1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 1,-1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,(-1, 0),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 0, 0),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 1, 0),axis=(0,1)).ravel(),
+        np.roll(diff_mean,(-1, 1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 0, 1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 1, 1),axis=(0,1)).ravel(),
+        \
+        np.roll(diff_mean,(-1,-2),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 0,-2),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 1,-2),axis=(0,1)).ravel(),
+        np.roll(diff_mean,(-1, 2),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 0, 2),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 1, 2),axis=(0,1)).ravel(),
+        \
+        np.roll(diff_mean,(-2,-1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,(-2, 0),axis=(0,1)).ravel(),
+        np.roll(diff_mean,(-2, 1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 2,-1),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 2, 0),axis=(0,1)).ravel(),
+        np.roll(diff_mean,( 2, 1),axis=(0,1)).ravel(),        
+        ]), axis=0)
+    diff_compare = np.reshape(diff_local_med[-ind_compare-1,:], diff_mean.shape)
+
+    # Generate mask
+    mask = diff_mean - diff_compare > thresh
+
+    # apply filtering
+    for ax, ay in tqdmnd(*(datacube.R_Nx,datacube.R_Ny), desc="Cleaning pixels", unit=" images"):
+        # Calculate local 3x3 median images
+        im_med = median_filter(
+            datacube.data[ax,ay,:,:],
+            size=3,
+            mode='nearest')
```

Comment:
for consistency with the `diff_compare` array, I would suggest changing to `mode='wrap'`

Reply:
What would we use wrap?  I think diff_compare should be changed!  The diffraction images are not periodic right?

Reply:
Changing `diff_compare` would also work - the main thing I want is for them to use the same boundary handling, whatever it is

Reply:
This is honestly a pretty minor point, I don't expect it will make much difference.  Using `mode='constant',cval=0` might make the most sense, given that these are most likely high |k| points in diffraction space....but I won't fiddle, this seems fine